### PR TITLE
Use pull-secret API and ask for copy-paste

### PR DESCRIPTION
**Pull request by gbraad**
Fixes: #99 and #100
Implemented according to: https://github.com/code-ready/crc/issues/2561
Remaining discussion: https://github.com/code-ready/crc/issues/2571 (what to do in case file is given)

---

When no pull-secret exists, a pullsecret dialog is shown. When pasted and confirmed, it is stored in the Credential Manager.

From Settings it is possible to change the pull-secret. Is stored when applied.

### CODE REVIEW

File: CRCTray.cs [hunk 1]
```diff
@@ -228,39 +229,28 @@ async private void StopMenu_Click(object sender, EventArgs e)
...
         {
             // Check using get-config if pullSecret is configured
-            var configs = await TaskHelpers.TryTaskAndNotify(TaskHandlers.ConfigView,
+            var pullsecret = await TaskHelpers.TryTaskAndNotify(TaskHandlers.GetPullSecret,
                 String.Empty,
                 String.Empty,
                 String.Empty);
 
-            if(configs == null)
+            if (!pullsecret)
```
Comment (anjannath) [inline CRCTray.cs:237]: This check is not working as expected, the pull-secret prompt always appears when you click on start even when its already saved in the credential store

Comment (anjannath) [reply CRCTray.cs:237]: If the pull secret is not stored the api returns a 404 http status not found

Comment (gbraad) [reply CRCTray.cs:237]: I'll have to check this, but this always returns `false`? The idea is that an exception returns a `default` typed result. I see that the 200, does not return any value, so this means the `bool` type will therefore have no value.

```
return getResultsForBasicCommand<bool>(BasicCommands.PullSecret);
```

**Comment by gbraad** [comment]
@anjannath did this work? else we have to take this into the next sprint

**Comment by anjannath** [comment]
@gbraad Now getting unhandled exceptions when clicking on start :(
```
Unhandled Exception: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.AggregateException: One or more errors occurred. ---> CRCTray.Communication.NotFoundException
   at CRCTray.Communication.DaemonCommander.<getResponse>d__17.MoveNext()
   --- End of inner exception stack trace ---
   at System.AggregateException.Handle(Func`2 predicate)
   at CRCTray.Helpers.TaskHelpers.<TryTaskAndNotify>d__0`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CRCTray.CRCTray.<StartMenu_Click>d__28.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_0(Object state)
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Delegate.DynamicInvokeImpl(Object[] args)
   at System.Windows.Forms.Control.InvokeMarshaledCallbackDo(ThreadMethodEntry tme)
   at System.Windows.Forms.Control.InvokeMarshaledCallbackHelper(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Windows.Forms.Control.InvokeMarshaledCallback(ThreadMethodEntry tme)
   at System.Windows.Forms.Control.InvokeMarshaledCallbacks()
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   at System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.DebuggableCallback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
   at System.Windows.Forms.UnsafeNativeMethods.DispatchMessageW(MSG& msg)
   at System.Windows.Forms.Application.ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(IntPtr dwComponentID, Int32 reason, Int32 pvLoopData)
   at System.Windows.Forms.Application.ThreadContext.RunMessageLoopInner(Int32 reason, ApplicationContext context)
   at System.Windows.Forms.Application.ThreadContext.RunMessageLoop(Int32 reason, ApplicationContext context)
   at CRCTray.Program.Main()
level=error msg="stdin is closed, shutdown..."
The program '[14176] crc-tray.exe' has exited with code 0 (0x0).
```

**Comment by gbraad** [comment]
Haven't been able to test/work on this.
Functionality isn't defective, just would be improved.

Defer to next sprint

File: CRCTray.cs (hunk 1 above)
Comment (gbraad) [reply CRCTray.cs:237]: AggregateException was thrown instead, containing the APINotFound

**Comment by anjannath** [comment]
@gbraad Will test this today

**Comment by gbraad** [comment]
Please check the alignment in the settings window; have a local change that makes no sense.

**Review by anjannath — APPROVED** [review]

**Comment by anjannath** [comment]
New settings windows looks like below:
image

**Comment by gbraad** [comment]
@anjannnath now to decide what to do on the macOS side about the 'file' option: https://github.com/code-ready/tray-macos/issues/120

**Pull request merged**